include/zephyr/init.h

```diff
@@ -96,7 +107,12 @@ struct init_entry {
 	 * If the init entry belongs to a device, this fields stores a
 	 * reference to it, otherwise it is set to NULL.
 	 */
-	const struct device *dev;
+	union {
+		const struct device *dev;
+#ifdef CONFIG_DEVICE_MUTABLE
+		struct device *dev_rw;
+#endif
+	};
```

Comment:
I think this should wait for all the changes scheduled for the init changes https://github.com/zephyrproject-rtos/zephyr/pull/61986, otherwise we have yet another thing on the table to fix.

Reply:
I don't see the other draft PR as necessarily blocking this, but I would be happy to minimize whatever might be necessary to fix. E.g. originally, I added the macros from mutable.h directly into device.h because I thought it might be nice to be slightly separate. I can move them back if it's simpler to be in the same file.

Reply:
after looking at it again, it seems that it should be easy to port, since the proposal removes devices from init.h.